components/camel-drill/src/main/java/org/apache/camel/component/drill/DrillEndpoint.java

```diff
@@ -0,0 +1,177 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.drill;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+import oadd.org.apache.commons.lang.StringUtils;
+
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.component.drill.DrillComponent.DrillConnectionMode;
+import org.apache.camel.impl.DefaultPollingEndpoint;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.springframework.jdbc.core.ColumnMapRowMapper;
+import org.springframework.jdbc.core.RowMapperResultSetExtractor;
+
+
+/**
+ * The drill component gives you the ability to quering into apache drill
+ * cluster.
+ */
+@UriEndpoint(scheme = "drill", title = "DRILL", syntax = "drill:host", producerOnly = true, label = "database,sql")
+public class DrillEndpoint extends DefaultPollingEndpoint {
+
+    @UriPath(description = "Host name or IP address")
+    @Metadata(required = "true")
+    private String host;
+    @UriParam(description = "Port number")
+    @Metadata(required = "false", defaultValue = "2181")
+    private Integer port = 2181;
+    @UriParam(description = "Drill directory", defaultValue = "")
+    private String directory = "";
+    @UriParam(defaultValue = "")
+    private String clusterId = "";
+    @UriParam(defaultValue = "zk")
+    private String mode = "zk";
```

Comment:
I think you can use the enum DrillConnectionMode here, and list possible values in UriParam